aip/client-libraries/4210.md

```diff
@@ -31,25 +35,90 @@ expectation.
 
 ## Guidance
 
+The general flow of code generation for client libraries and supporting code is
+outlined below.
+
+```graphviz
+digraph {
+  node [ style="filled,solid" shape=box fontname="Roboto" ];
+  splines=ortho;
+  nodesep=0.3;
+  center=true;
+
+  proto [ label="API Protobuf\nDescriptors" shape=rectancle fillcolor=aliceblue ];
+
+  subgraph cluster_code_generators {
+    rank = same;
+    style = filled;
+    fillcolor = lightgrey;
+    node [ shape=oval ];
+
+    protobuf [ label="protobuf\ngenerator" fillcolor=deepskyblue3 ];
+    grpc [ label="gRPC\ngenerator" fillcolor=gold3 ];
+    gapic [ label="GAPIC\ngenerator" fillcolor=darkseagreen ];
+
+  }
+
+  proto -> protobuf;
+  proto -> grpc;
+  proto -> gapic;
+
+  subgraph cluster_generated_code {
+    rank = same;
+    style = filled;
+    fillcolor = lightgrey;
+    node [ shape=rectangle ];
+
+
+    protobuf_output [ label="Message & Enum\nCode" fillcolor=deepskyblue3 ];
+    grpc_output [ label="Server & Client\nStubs" fillcolor=gold3 ];
+    gapic_output [ label="Google API\nClient" fillcolor=darkseagreen ];
+
+  }
+  
+  protobuf -> protobuf_output;
+  grpc -> grpc_output;
+  gapic -> gapic_output;
+
+  assembly [ label="Package\nassembly" shape=oval fillcolor=aliceblue ];
+  
+  protobuf_output -> assembly
+  grpc_output -> assembly
+  gapic_output -> assembly
+
+  assembled_package [ label="Package of\ngenerated code" fillcolor=aliceblue ];
+
+  assembly -> assembled_package
+}
+```
+
+The following sections focus on the "GAPIC generator" in the above diagram.
+
 ### Protobuf plugins
 
-Code generators **must** be implemented as plugins to `protoc`, the protocol
-buffer compiler. The [plugin system][0] allows plugins to be written in any
-language, and plugins **should** ordinarily be written in the language being
-targeted, in order to take advantage of in-language tooling, and to ensure that
-experts in the target environment are able to meaningfully contribute.
+The protobuf compiler, `protoc`, supports a [plugin system][0] for code
+generation. The [plugin system][0] allows plugins to be written _in_ and _for_
```

Comment:
nit: the second `plugin system` does not need to be a link.

Reply:
done